### PR TITLE
fix(ext-plugin): avoid using 0 as the default http status code

### PR DESCRIPTION
**Pull request by tzssangglass**
### What this PR does / why we need it:

If the plugin-runner sends a stop request but does not set an Http Status Code, APISIX will use 0 as the default status code, causing the current request to run later phases.

see: https://github.com/apache/apisix-java-plugin-runner/issues/55
### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: t/plugin/ext-plugin/http-req-call.t [hunk 1]
```diff
@@ -124,7 +124,28 @@ X-Req: bar
 
 
 
-=== TEST 3: check input
+=== TEST 3: stop without setting status code
```
Comment (spacewander) [inline t/plugin/ext-plugin/http-req-call.t:127]: Please add new test at the end of file

Comment (tzssangglass) [reply t/plugin/ext-plugin/http-req-call.t]: update

File: t/plugin/ext-plugin/http-req-call.t [hunk 2]
```diff
@@ -517,3 +517,24 @@ GET /hello
...
+--- extra_stream_config
+    server {
+        listen unix:$TEST_NGINX_HTML_DIR/nginx.sock;
+
+        content_by_lua_block {
+            local ext = require("lib.ext-plugin")
+            ext.go({stop = true, stop_no_code = true})
+        }
+    }
+--- error_code: 200
```
Comment (spacewander) [inline t/plugin/ext-plugin/http-req-call.t:537]: We already check if error_code == 200 by default

File: t/lib/ext-plugin.lua [hunk 3]
```diff
@@ -185,7 +185,9 @@ function _M.go(case)
             local vec = builder:EndVector(len)
 
             http_req_call_stop.Start(builder)
-            http_req_call_stop.AddStatus(builder, 405)
+            if case.stop_no_code ~= true then
```
Comment (spacewander) [inline t/lib/ext-plugin.lua:188]:
```suggestion
            if case.check_default_status ~= true then
```

Comment (tzssangglass) [reply t/lib/ext-plugin.lua]: update

**Review by spacewander — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**